### ISSUE TITLE
[Feature] Metrics should work on ephemeral models for now

### ISSUE THREAD

**Issue by callum-mcdata** [dbt-labs/dbt_metrics#198]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Exactly as it says! This is a common pattern for folks who are wanting to join models together without materializing them in the data warehouse. 

There are some issues here around information in the manifest and what we can reference. There **might** be a way we can resolve this in `gen_base_query`.

### Describe alternatives you've considered

Not doing this - at the end of the day its a stopgap for entities and joins.

### Who will this benefit?

Anyone who wants to use joins without view/table proliferation.

### Are you interested in contributing this feature?

Yes

**Comment by bennieregenold7** [comment]
Using ephemeral models as the "object" that BI tools use feels like an anti-pattern to me. Our [recommendation on materializations](https://docs.getdbt.com/docs/build/materializations#table) says that you should use a table for anything that's used by end users/tools. Additionally, our [recommendation on ephemeral models](https://docs.getdbt.com/docs/build/materializations#ephemeral) says that these should not be queried directly, and should not appear at the end of a DAG.

Based on these recommendations, and what I've seen doing consulting work, I don't think we should support ephemeral models in the semantic layer. If an analyst needs to join models together it should at the very least be a view, but more likely a table. 

When using ephemeral models you tend to run into memory issues in the warehouse. This is especially true if you're joining two final fact or dimension tables that will likely be very large (assuming you have the detailed grain needed to summarize in the semantic layer).

**Comment by callum-mcdata** [comment]
@bennieregenold7 Agreed - this was a stopgap solution to a debate we were having around entities. I think we're more comfortable saying we want to work on joins and don't need this in the interim. Going to close this issue.
